### ISSUE TITLE
Update RecurringHostedService example to include logger for v9.4+

### ISSUE THREAD

**Issue by ronaldbarendse** [umbraco/UmbracoDocs#3883]
## What type of issue is it?
Wrong documentation: the example code is calling a base constructor that has been obsoleted in 9.4+

## What article/section is this about?
[Scheduling with RecurringHostedServiceBase](https://our.umbraco.com/Documentation/Reference/Scheduling/)

## Describe the issue
Umbraco 9.4.0 includes two PRs (https://github.com/umbraco/Umbraco-CMS/pull/12153 and https://github.com/umbraco/Umbraco-CMS/pull/12161) that fixed a critical issue when using the `RecurringHostedServiceBase`, but now requires inherited classes to pass in an `ILogger` instance to a new base constructor. The previous constructor got obsoleted, but used a fallback to automatically get an `ILogger` instance that didn't work as expected, so 9.4.1 got released on the same day (that includes PR https://github.com/umbraco/Umbraco-CMS/pull/12172).

TL;DR The current `RecurringHostedServiceBase` example code doesn't work on 9.4.0 and although this got fixed in 9.4.1, users targeting 9.4+ should use the new constructor.

I'm not sure what makes sense regarding updating the documentation: include a note/code comment users should use a different base constructor on 9.4+ or create a version specific page for this?

**Comment by sofietoft** [comment]
Hi @ronaldbarendse !

Thanks for the detailed description of the case here.
I think how we handle this comes down to how confident we are in terms of people always running the latest and greatest version 😅 

We should try to avoid creating multiple versions of articles if possible. So, if the amount of users implicated isn't too high, the current article should be updated to contain the correct code and samples - and that kinda sounds like the case here.
And then we could add a note for those who haven't yet upgraded to 9.4+.

Does that make sense? And what do you think of that solution?

**Comment by sofietoft** [comment]
Hi @ronaldbarendse !

It's been a while since I've heard back from you regarding this one.
Did you see me message ☝️ 

Is the Issue reported here still relevant, and what do we need in order to get it fixed?
